### DISCUSSION TITLE
Mobile app can NOT display wiki content from render/content column field

### DISCUSSION THREAD

**Discussion by aapwikip** [requarks/wiki#6840] [Help / Questions]
Hi

We are looking to develop an android app for our wiki website. Any reference regarding this will appreciated

Specifically we are looking for below info on fetching wiki content from PostGreSql Database?
a. Name of different tables
b. Format of the table
c. Sample Queries to fetch data from these table

UPDATE: I am able to access 2 column representing Page Contents 'Content' and 'render' but I am unable to display it properly. Probably due to missing css/style info from these DB fields

Regards

**Comment by NGPixel** [comment]
Have you tried browsing your DB? All table names are self-explanatory and you can see the format with any database management tool...

**Reply by aapwikip** [reply]
Yes and I found a better solution i.e. GraphQl over Direct DB access

Either Way, I am able to access 2 column representing Page Contents 'Content' and 'render' but I am unable to display it properly.
There are huge differences in formatting and styling e.g. there are no CSS/Styling information with these columns

My understanding is that standard markdown render may not work for 'Content' column as these are some custom renderings. I was hoping to get Raw Html with head/styles info etc to just display it as it is but 'render' content is not as expected

Please guide us about how to display the content correctly in an app

**Reply by rrg92** [reply]
You can use render column, and load into in a DOM Parser and then convert html that best fits your app.

When a page is created/modified, wiki core will render it, getting the content (the output of editor or data sent in api) and generate a html that will be displayed.

Every time you acess a page, you ate seing the render column, so, in your mobile, you use this to render.

But rendering process is made to render in browsers, and depends on all libraries loaded to page, so, just getting render and displaying is not sufficiente. You all need all css too.

Using render column allows you get a dom structure that you can parse and convert yo your app. For example, you can get all h1 elements and convert to something that your app will show as title. Also, you can use content property of page and do a new rendering of you own (i preferr that). This will create a need of maintain a new rendering process... but this is not different with render property. The main diff is that using content, you must parse user direct data and using render you parse what wiki core want show
